Add support to configure network_cli terminal related options (#60086)
ansible/ansible: docs/docsite/rst/network/user_guide/network_debug_troubleshooting.rst

   connect_retry_timeout = 30

> Timeout issue due to platform specific login menu with network_cli connection type

Reviewer: ```suggestion
Timeout issue due to platform specific login menu with ``network_cli`` connection type
```

----------------------------------------------------------------------------------

> Starting in Ansible 2.9 version onwards the terminal plugin supports configuration options

Reviewer: ```suggestion
In Ansible 2.9 and later,  the terminal plugin supports configuration options
```

> to handle the platform specific login menu. These options can be set at global or group/host or as

Reviewer: ```suggestion
to handle the platform specific login menu. These options can be set as global, group/host or 
```

tasks variables.

> Example Handle login menu with host variables

Reviewer: ```suggestion
Example: Handle login menu with host variables
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

.. code-block:: console

    $cat host_vars/<hostname>.yaml
    ---
    ansible_terminal_initial_prompt:
      - "Connect to a host"
    ansible_terminal_initial_answer:
      - "3"

> Example Handle remote host login menu with host variables

Reviewer: ```suggestion
Example: Handle remote host login menu with host variables
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

.. code-block:: console

    $cat host_vars/<inventory-hostname>.yaml
    ---
    ansible_terminal_initial_prompt:
      - "Press any key to enter main menu"
      - "Connect to a host"
    ansible_terminal_initial_answer:
      - "\\r"
      - "3"
    ansible_terminal_initial_prompt_checkall: True

To handle multiple login menu prompts the values of ``ansible_terminal_initial_prompt`` and
``ansible_terminal_initial_answer`` should be a list and the prompt sequence should match the

> answer sequence and value of ``ansible_terminal_initial_prompt_checkall`` should be set to

Reviewer: Can you turn this into a bullet list? To handle multiple login menu prompts:

* The values of  x an y should be a list
* The prompt sequence...
* The value of ...

> ``True``. If all the prompts in sequence are not received from remote host it will result in

Reviewer: ```suggestion
``True``. If all the prompts in sequence are not received from remote host it will result in a
```

   connect_retry_timeout = 30

Timeout issue due to platform specific login menu with ``network_cli`` connection type
--------------------------------------------------------------------------------------

In Ansible 2.9 and later, the network_cli connection plugin configuration options are added
to handle the platform specific login menu. These options can be set as group/host or tasks
variables.

Example: Handle single login menu prompts with host variables

.. code-block:: console

    $cat host_vars/<hostname>.yaml
    ---
    ansible_terminal_initial_prompt:
      - "Connect to a host"
    ansible_terminal_initial_answer:
      - "3"

Example: Handle remote host multiple login menu prompts with host variables

.. code-block:: console

    $cat host_vars/<inventory-hostname>.yaml
    ---
    ansible_terminal_initial_prompt:
      - "Press any key to enter main menu"
      - "Connect to a host"
    ansible_terminal_initial_answer:
      - "\\r"
      - "3"
    ansible_terminal_initial_prompt_checkall: True

To handle multiple login menu prompts:

> * The values of ``ansible_terminal_initial_prompt`` and ``ansible_terminal_initial_answer`` should be a list.

Reviewer: ```suggestion

* The values of ``ansible_terminal_initial_prompt`` and ``ansible_terminal_initial_answer`` should be a list.
```

   buffer_read_timeout = 2

This timer delay per command executed on remote host can be disabled by setting the value to zero.

> Task failure due to mismatched error regex within command response using network_cli connection type

Reviewer: ```suggestion
Task failure due to mismatched error regex within command response using ``network_cli`` connection type
```

----------------------------------------------------------------------------------------------------

> Starting in Ansible 2.9 version onwards the terminal plugin supports configuration options

Reviewer: ```suggestion
In Ansible 2.9 and later the terminal plugin supports configuration options
```

to handle the stdout and stderr regex to identify if the command execution response consist

> of normal response or error response. These options can be set group/host variables or as

Reviewer: ```suggestion
of a normal response or an error response. These options can be set group/host variables or as
```

tasks variables.

> Example For mismatched error response

Reviewer: ```suggestion
Example: For mismatched error response
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

.. code-block:: yaml

  - name: fetch logs from remote host
    ios_command:
      commands:
        - show logging

Playbook run output:

.. code-block:: console

  TASK [first fetch logs] ********************************************************
  fatal: [ios01]: FAILED! => {
      "changed": false,
      "msg": "RF Name:\r\n\r\n <--nsip-->
             \"IPSEC-3-REPLAY_ERROR: Test log\"\r\n*Aug  1 08:36:18.483: %SYS-7-USERLOG_DEBUG:
              Message from tty578(user id: ansible): test\r\nan-ios-02#"}

Suggestions to resolve:

> Modify the error regex for individual task

Reviewer: ```suggestion
Modify the error regex for individual task.
```

.. code-block:: yaml

  - name: fetch logs from remote host
    ios_command:
      commands:
        - show logging
    vars:
      ansible_terminal_stderr_re:
        - pattern: 'connection timed out'
          flags: 're.I'

> The terminal plugin regex options ``ansible_terminal_stderr_re`` and ``ansible_terminal_stdout_re`` has

Reviewer: ```suggestion
The terminal plugin regex options ``ansible_terminal_stderr_re`` and ``ansible_terminal_stdout_re`` have
```

> ``pattern`` and ``flags`` as keys. The value of ``flags`` key should be a value that is accepted by

Reviewer: ```suggestion
``pattern`` and ``flags`` as keys. The value of the ``flags`` key should be a value that is accepted by
```

> ``re.compile`` python method.

Reviewer: ```suggestion
the ``re.compile`` python method.
```
